packages/app/src/cli/services/event/trigger.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  topicPrompt,
+  addressPrompt,
+  apiVersionPrompt,
+  localPortPrompt,
+  localUrlPathPrompt,
+  sharedSecretPrompt,
+  deliveryMethodPrompt,
+} from '../../prompts/event/trigger.js'
+import {api, http, session} from '@shopify/cli-kit'
+
+export interface SamplePayloadSchema {
+  sendSampleWebhook: {
+    samplePayload: string
+    headers: string
+    success: boolean
+    userErrors: UserErrors[]
+  }
+}
+
+export interface UserErrors {
+  message: string
+  fields: string[]
+}
+
+export const DELIVERY_METHOD = {
+  LOCALHOST: 'localhost',
+  HTTP: 'http',
+  PUBSUB: 'google-pub-sub',
+  EVENTBRIDGE: 'event-bridge',
+}
+
+const sendSampleWebhookMutation = `
+  mutation samplePayload($topic: String!, $api_version: String!, $address: String!, $delivery_method: String!, $shared_secret: String!) {
+    sendSampleWebhook(input: {topic: $topic, apiVersion: $api_version, address: $address, deliveryMethod: $delivery_method, sharedSecret: $shared_secret}) {
+        samplePayload
+        success
+        headers
+        userErrors {
+          message
+        }
+    }
+  }
+`
+
+export async function requestSample(
+  topic: string,
+  apiVersion: string,
+  deliveryMethod: string,
+  address: string,
+  sharedSecret: string,
+) {
+  const token = await session.ensureAuthenticatedPartners()
+
+  const variables = {
+    topic,
+    api_version: apiVersion,
+    address,
+    delivery_method: deliveryMethod,
+    shared_secret: sharedSecret,
+  }
+
+  const {sendSampleWebhook: result}: SamplePayloadSchema = await api.partners.request(
+    sendSampleWebhookMutation,
+    token,
+    variables,
+  )
+
+  return result
+}
+
+export async function sendLocal(address: string, body: string, headers: string) {
+  const options = {
+    method: 'POST',
+    body,
+    headers: {
+      'Content-Type': 'application/json',
+      ...JSON.parse(headers),
+    },
+  }
+  const response = await http.fetch(address, options)
+  return response.status === 200
+}
+
+export interface TestWebhookOptions {
+  topic: string
+  apiVersion: string
+  deliveryMethod: string
+  sharedSecret: string
+  localhostPort: string
+  localhostUrlPath: string
+  address: string
+}
+
+export interface TestWebhookFlags {
+  topic?: string
+  apiVersion?: string
+  deliveryMethod?: string
+  port?: string
+  address?: string
+  urlPath?: string
+}
+
+export async function collectCliOptions(flags: TestWebhookFlags): Promise<TestWebhookOptions> {
+  const options: TestWebhookOptions = {
+    topic: await collectParam(flags.topic as string, topicPrompt),
+    apiVersion: await collectParam(flags.apiVersion as string, apiVersionPrompt),
+    sharedSecret: await collectParam(process.env.SHOPIFY_FLAG_SHARED_SECRET as string, sharedSecretPrompt),
+    deliveryMethod: await collectParam(flags.deliveryMethod as string, deliveryMethodPrompt),
```

Comment:
I've been trying to prompt all of them at once, but I'm not sure I can. I renamed `collectParam` to something more meaningful like `useFlagOrPrompt`.  For each one of those, I only prompt if they have not been passed as flags, so the ui.prompt array is not fixed for me.

I have also moved the logic so that prompts are only requested after checking that none of the passing flags are wrong